src/transport/ExchangeMgr.h

```diff
@@ -0,0 +1,270 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines the classes corresponding to CHIP Exchange management.
+ *
+ */
+
+#ifndef EXCHANGE_MGR_H
+#define EXCHANGE_MGR_H
+
+#include <support/DLLUtil.h>
+#include <system/SystemTimer.h>
+#include <transport/SecureSessionMgr.h>
+
+#define EXCHANGE_CONTEXT_ID(x) ((x) + 1)
```

Comment:
this didn't make sense as a member of ExchangeContext?

Reply:
Those for ExchangeManager, ExchangeContext and ExchangeManager are defined in the same header file since they are closely coupled together.

SecurePairingSession.h and SecureSessionMgr.h also adopts this pattern which glues several closely coupled classes into one header file.

Reply:
Split the ExchangeContext to a separate interface